src/aiida_common_workflows/workflows/em.py

```diff
@@ -0,0 +1,134 @@
+"""Equation of state workflow that can use any code plugin implementing the common relax workflow."""
+import inspect
+
+from aiida import orm
+from aiida.common import exceptions
+from aiida.engine import WorkChain, append_
+from aiida.plugins import WorkflowFactory
+
+from aiida_common_workflows.workflows.relax.generator import ElectronicType, RelaxType, SpinType
+from aiida_common_workflows.workflows.relax.workchain import CommonRelaxWorkChain
+
+
+def validate_inputs(value, _):
```

Comment:
I tested running this workchain using relax.cp2k as `sub_process_class`. It does not fail till reaching line 108. I would add here a check that explicitly controls that the `fixed_total_cell_magnetization` is supported for the selected sub_process_class. The code here would fail only if the user sets `generator_inputs.fixed_total_cell_magnetization`, which I do not think he/she should, see next comment.

Reply:
Very helpful. Indeed, I overlooked this and only checked that it fails (but only at runtime). This is now properly handled during the validation, so that the workflow is not submitted.